src/main/java/net/openhft/chronicle/wire/GenerateMethodWriter.java

```diff
@@ -459,14 +460,18 @@ private CharSequence createMethod(SortedSet<String> importSet, final Method dm,
                     ".update(\"" + dm.getName() + "\", " + name + ")) return" + returnDefault(returnType) + ";\n");
         }
 
+        body.append("MarshallableOut out = this.out.get();\n");
+        if (dm.getDeclaringClass() == Syncable.class) {
+            body.append(Syncable.class.getName()).append(".syncIfAvailable(out);\n");
```

Comment:
what does this do different?

Reply:
If the interface for the method writer extends Syncable and you call sync() it will call the same on the underlying MarshallableOut if it supports Syncable before writing the `sync: ""`
This way, the caller can determine programmatically when sync() should be performed and the reader can determine at what point sync() was called upstream.
Forough has an issue to create a test in services with a demo.